### ISSUE TITLE
Output size of ray tracing, compressing?

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/gammasim-tools#80]
I know that file format is something undecided on. But would it be possible to compress at least all the ascii output at the end of the analysis?

Just zipping the output from derive_mirror_rnda reduces the storage footprint by a factor larger than 4.

**Comment by orelgueta** [comment]
I played around a little bit and we could actually gzip the sim_telarray files before the analysis as well. The extra compression adds to the run time though (~8 minutes vs ~3 minutes, running the example once each time). 
Not sure if it's worth doing before the analysis or after (probably after), I was just curious about it cause I know both sim_telarray and python should support gzip files ~natively.

I can open a branch with my changes if anyone is interested in this.

**Comment by GernotMaier** [comment]
Definitely a  minor issue, but saving uncompressed ascii is not very efficient.

**Comment by orelgueta** [comment]
Well, minor, but important nonetheless I would say. 
It won't be affected by a choice of file format, cause at least for the first part we are limited by sim_telarray writing to ascii (I doubt Konrad would want to change this software to write to HDF5). 
We then have to then deal with these ascii files. As I mentioned, I added a step that compresses the files before we analyse them. It's clean-ish, but comes with a performance penalty. A more efficient solution (but not as pretty) is to compress the output directory after the application finishes the run. @RaulRPrado, what do you think?

Oh, one more thing, if we write the output to a DB and delete it from the local machine, that might solve the problem as well. Though I don't know if we will do that for these applications.

**Comment by GernotMaier** [comment]
A clear description of what we write to disk is something we have to come up with at some point. Clearly, we don't need to save every photon created in the ray tracing, but obviously everything needed to re-run the analysis (parameters; random seeds).

**Comment by orelgueta** [comment]
> everything needed to re-run the analysis (parameters; **random seeds**).

Right, that was the other point which I forgot to write here and why opened #91.

**Comment by GernotMaier** [comment]
@orelgueta  where did you add the zipping of the output?

Was running applications/derive_mirror_rnda.py and it produces ~1GB of photon lists (e.g., files like photons-North-MST-FlashCam-D-d0.0-za20.0-off0.000_mirror185_derive_mirror_rnda.lis)

Could we add a run option allows to select:

- full output with all photons lists (but zipped please)
- only the final figures and numbers (but keep everything needed to rerun the analysis)

The large number of files produced worries: think about we are running the every night..maybe we can find a way to put everything into one single file as output (but this is a discussion beyond this issue, and probably lower priority)

**Comment by orelgueta** [comment]
Added it only in my local version where I played around with the different options. I did not open a branch to include these changes since I was waiting for @RaulRPrado to tell us his opinion regarding zipping before/after running (at the moment I vote for after).
Not sure if I can find that code again, but I can try. Either way, it wasn't too complicated and can be done again.

I agree with your suggestion of adding it as a user option (with the default of deleting the photon files).
